service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -94,13 +101,42 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = getCaseData(callbackRequest.getCaseDetails());
         CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
 
-        if (isRemovingNewSDO(caseData, caseDataBefore)) {
+        Optional<StandardDirectionOrder> removedSDO = getRemovedSDO(caseData, caseDataBefore);
+        Optional<CaseManagementOrder> removedCMO = getRemovedCMO(caseData, caseDataBefore);
+
+        if (removedSDO.isPresent()) {
             publishEvent(new PopulateStandardDirectionsEvent(callbackRequest));
+            publishEvent(new StandardDirectionsOrderRemovedEvent(
+                caseData, removedSDO.map(StandardDirectionOrder::getRemovalReason).orElse(null)));
```

Comment:
Why the orElse(null). Removal reason is mandatory and should always be there.

Reply:
I had to set the default value. As discussed, I've changed it to an empty string now